executor/cluster_reader_test.go

```diff
@@ -0,0 +1,217 @@
+		"tikv key2.nest2 n-value2",
+		"tikv key1 value1",
+		"tikv key2.nest1 n-value1",
+		"tikv key2.nest2 n-value2",
+		"pd key1 value1",
+		"pd key2.nest1 n-value1",
+		"pd key2.nest2 n-value2",
+		"pd key1 value1",
+		"pd key2.nest1 n-value1",
+		"pd key2.nest2 n-value2",
+		"pd key1 value1",
+		"pd key2.nest1 n-value1",
+		"pd key2.nest2 n-value2",
+	))
+	warnings := tk.Se.GetSessionVars().StmtCtx.GetWarnings()
+	c.Assert(len(warnings), Equals, 0, Commentf("unexpected warnigns: %+v", warnings))
+	c.Assert(requestCounter, Equals, int32(9))
+
+	var cases = []struct {
+		sql      string
+		reqCount int32
+	}{
+		{
+			sql:      "select * from information_schema.cluster_config where type='tikv'",
+			reqCount: 3,
+		},
+		{
+			sql:      "select * from information_schema.cluster_config where type='pd'",
+			reqCount: 3,
+		},
+		{
+			sql:      "select * from information_schema.cluster_config where type='tidb'",
+			reqCount: 3,
+		},
+		{
+			sql:      "select * from information_schema.cluster_config where 'tidb'=type",
+			reqCount: 3,
+		},
+		{
+			sql:      "select * from information_schema.cluster_config where type in ('tidb', 'tikv')",
+			reqCount: 6,
+		},
+		{
+			sql:      "select * from information_schema.cluster_config where type in ('tidb', 'tikv', 'pd')",
+			reqCount: 9,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where address='%s'`,
+				testServers[0].address),
+			reqCount: 3,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type='tidb' and address='%s'`,
+				testServers[0].address),
+			reqCount: 1,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type in ('tidb', 'tikv') and address='%s'`,
+				testServers[0].address),
+			reqCount: 2,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type in ('tidb', 'tikv') and address in ('%s', '%s')`,
+				testServers[0].address, testServers[0].address),
+			reqCount: 2,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type in ('tidb', 'tikv') and address in ('%s', '%s')`,
+				testServers[0].address, testServers[1].address),
+			reqCount: 4,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type in ('tidb', 'tikv') and type='pd' and address in ('%s', '%s')`,
+				testServers[0].address, testServers[1].address),
+			reqCount: 0,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type in ('tidb', 'tikv') and address in ('%s', '%s') and address='%s'`,
+				testServers[0].address, testServers[1].address, testServers[2].address),
+			reqCount: 0,
+		},
+		{
+			sql: fmt.Sprintf(`select * from information_schema.cluster_config where type in ('tidb', 'tikv') and address in ('%s', '%s') and address='%s'`,
+				testServers[0].address, testServers[1].address, testServers[0].address),
+			reqCount: 2,
+		},
+	}
+
```

Comment:
Please add test cases like：
`select * from information_schema.cluster_config where type=xx or address=xx`
`select * from information_schema.cluster_config where type=xx or type=xx`
`select * from information_schema.cluster_config where type=xx and type=xx`
`select * from information_schema.cluster_config`